Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TimeOffController.cs

```diff
@@ -244,29 +244,52 @@ internal async Task ProcessTimeOffsAsync(string isRequestFromLogicApp)
             List<PayCodeToTimeOffReasonsMappingEntity> timeOffRequestsPayCodeList,
             List<GlobalTimeOffRequestItem> globalTimeOffRequestDetails)
         {
+            this.telemetryClient.TrackTrace($"ProcessTimeOffEntitiesBatchAsync started.");
+
             foreach (var item in processKronosUsersQueueInBatch)
             {
                 foreach (var timeOffRequestItem in timeOffResponseDetails.Where(x => x.Employee.PersonIdentity.PersonNumber == item.KronosPersonNumber))
                 {
                     if (timeOffRequestItem.StatusName.Equals(ApiConstants.ApprovedStatus, StringComparison.Ordinal))
                     {
+                        this.telemetryClient.TrackTrace($"ProcessTimeOffEntitiesBatchAsync look up count: {lookUpData.Count} ");
+
                         if (lookUpData.Count == 0)
                         {
+                            // Performing the query using the RowKey as the paycode name from Kronos; and the PartitionKey as the TeamID.
                             var timeOffReasonId = timeOffReasons.
                                 Where(t => t.RowKey == timeOffRequestItem.TimeOffPeriods.TimeOffPeriod.PayCodeName && t.PartitionKey == item.ShiftTeamId).FirstOrDefault();
-                            timeOffNotFoundList.Add(timeOffRequestItem);
-                            userModelNotFoundList.Add(item);
-                            kronosPayCodeList.Add(timeOffReasonId);
+                            this.telemetryClient.TrackTrace($"ProcessTimeOffEntitiesBatchAsync PaycodeName : {timeOffRequestItem.TimeOffPeriods.TimeOffPeriod.PayCodeName}, Kronos request ID : {timeOffRequestItem.Id}");
+                            
+                            // The null check is being performed on the timeOffReasonId because there may be chances where
```

Comment:
Do you mean "chances" here, or "cases"?

Reply:
@aosolis I mean chances here, and I'm clarifying the comments shortly